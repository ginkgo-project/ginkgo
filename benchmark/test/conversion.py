#!/usr/bin/env python3
import test_framework

# check that all input modes work:
# parameter
test_framework.compare_output(
    ["-input", '[{"size": 100, "stencil": "7pt"}]', "-formats", "coo,csr"],
    expected_stdout="conversion.simple.stdout",
    expected_stderr="conversion.simple.stderr",
)

# stdin
test_framework.compare_output(
    ["-formats", "coo,csr"],
    expected_stdout="conversion.simple.stdout",
    expected_stderr="conversion.simple.stderr",
    stdin='[{"size": 100, "stencil": "7pt"}]',
)

# input file
test_framework.compare_output(
    [
        "-input",
        str(test_framework.sourcepath / "input.mtx.json"),
        "-formats",
        "coo,csr",
    ],
    expected_stdout="conversion.simple.stdout",
    expected_stderr="conversion.simple.stderr",
)

# check that all conversions work
test_framework.compare_output(
    [
        "-input",
        '[{"size": 100, "stencil": "7pt"}]',
        "-formats",
        "coo,csr,ell,sellp,hybrid",
    ],
    expected_stdout="conversion.all.stdout",
    expected_stderr="conversion.all.stderr",
)

# profiler annotations
test_framework.compare_output(
    [
        "-input",
        '[{"size": 100, "stencil": "7pt"}]',
        "-formats",
        "coo,csr",
        "-profile",
        "-profiler_hook",
        "debug",
    ],
    expected_stdout="conversion.profile.stdout",
    expected_stderr="conversion.profile.stderr",
)
