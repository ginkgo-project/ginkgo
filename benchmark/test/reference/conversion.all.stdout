
[
    {
        "size": 125,
        "stencil": "7pt",
        "conversions": {
            "coo-csr": {
                "time": 1.0,
                "repetitions": 10,
                "completed": true
            },
            "coo-ell": {
                "completed": false,
                "error": ""
            },
            "coo-sellp": {
                "completed": false,
                "error": ""
            },
            "coo-hybrid": {
                "completed": false,
                "error": ""
            },
            "csr-coo": {
                "time": 1.0,
                "repetitions": 10,
                "completed": true
            },
            "csr-ell": {
                "time": 1.0,
                "repetitions": 10,
                "completed": true
            },
            "csr-sellp": {
                "time": 1.0,
                "repetitions": 10,
                "completed": true
            },
            "csr-hybrid": {
                "time": 1.0,
                "repetitions": 10,
                "completed": true
            },
            "ell-coo": {
                "completed": false,
                "error": ""
            },
            "ell-csr": {
                "time": 1.0,
                "repetitions": 10,
                "completed": true
            },
            "ell-sellp": {
                "completed": false,
                "error": ""
            },
            "ell-hybrid": {
                "completed": false,
                "error": ""
            },
            "sellp-coo": {
                "completed": false,
                "error": ""
            },
            "sellp-csr": {
                "time": 1.0,
                "repetitions": 10,
                "completed": true
            },
            "sellp-ell": {
                "completed": false,
                "error": ""
            },
            "sellp-hybrid": {
                "completed": false,
                "error": ""
            },
            "hybrid-coo": {
                "completed": false,
                "error": ""
            },
            "hybrid-csr": {
                "time": 1.0,
                "repetitions": 10,
                "completed": true
            },
            "hybrid-ell": {
                "completed": false,
                "error": ""
            },
            "hybrid-sellp": {
                "completed": false,
                "error": ""
            }
        }
    }
]
